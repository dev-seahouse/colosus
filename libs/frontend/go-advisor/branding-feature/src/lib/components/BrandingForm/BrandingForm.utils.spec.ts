import {
  getBrandingDefaultValues,
  hasLogoChanged,
  isNewLogo,
} from './BrandingForm.utils';

describe('BrandingForm.utils', () => {
  describe('getBrandingDefaultValues', () => {
    it('should return branding default values', () => {
      expect(
        getBrandingDefaultValues({
          logoUrl: null,
          brandColor: '#000000',
          headerBgColor: '#000000',
          tradeName: 'Wealth Avenue',
        })
      ).toMatchSnapshot();
    });
  });

  describe('hasLogoChanged', () => {
    it('should return true if oldLogo is null', () => {
      expect(hasLogoChanged(null, { url: 'https://www.google.com' })).toEqual(
        true
      );
    });

    it('should return true if newLogo is null', () => {
      expect(hasLogoChanged({ url: 'https://www.google.com' }, null)).toEqual(
        true
      );
    });

    it('should return false if both logos are null', () => {
      expect(hasLogoChanged(null, null)).toEqual(false);
    });

    it('should return true if logo urls do not match', () => {
      expect(
        hasLogoChanged(
          { url: 'https://www.google.com' },
          { url: 'https://www.bing.com' }
        )
      ).toEqual(true);
    });

    it('should return false if logo urls match', () => {
      expect(
        hasLogoChanged(
          { url: 'https://www.google.com' },
          { url: 'https://www.google.com' }
        )
      ).toEqual(false);
    });
  });

  describe('isNewLogo', () => {
    it('should return true if oldLogo is null and newLogo has value', () => {
      expect(isNewLogo(null, { url: 'https://www.google.com' })).toEqual(true);
    });

    it('should return true if oldLogo url and newLogo url do not match', () => {
      expect(
        isNewLogo(
          { url: 'https://www.google.com' },
          { url: 'https://www.bing.com' }
        )
      ).toEqual(true);
    });

    it('should return false if oldLogo url exists and newLogo url does not exist', () => {
      expect(isNewLogo({ url: 'https://www.google.com' }, null)).toEqual(false);
    });
  });
});
