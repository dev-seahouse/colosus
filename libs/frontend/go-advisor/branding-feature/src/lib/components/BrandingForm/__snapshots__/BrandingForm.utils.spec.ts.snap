// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`BrandingForm.utils > getBrandingDefaultValues > should return branding default values 1`] = `
{
  "brandColor": "#000000",
  "headerBgColor": "#000000",
  "logo": null,
  "tradeName": "Wealth Avenue",
}
`;
